src/libs/elektra/opmphm.c

```diff
@@ -438,52 +431,50 @@ OpmphmGraph * opmphmGraphNew (Opmphm * opmphm, uint8_t r, size_t n, double c)
 		opmphm->rUniPar = r;
 	}
 	// calculate opmphm->componentSize, number of elements in one part of r-uniform r-partite hypergraph
-	size_t componentSize = (c * n / opmphm->rUniPar) + 1;
+	opmphm->componentSize = (c * n / opmphm->rUniPar) + 1;
 	// mallocs
 	OpmphmGraph * graph = elektraMalloc (sizeof (OpmphmGraph));
 	if (!graph)
 	{
+		opmphm->componentSize = 0;
 		return NULL;
 	}
 	graph->edges = elektraMalloc (n * sizeof (OpmphmEdge));
 	if (!graph->edges)
 	{
+		opmphm->componentSize = 0;
 		elektraFree (graph);
 		return NULL;
 	}
-	uint32_t * vertices = elektraMalloc (n * opmphm->rUniPar * sizeof (uint32_t));
-	if (!vertices)
+	graph->vertices = elektraCalloc (opmphm->componentSize * opmphm->rUniPar * sizeof (OpmphmVertex));
+	if (!graph->vertices)
 	{
+		opmphm->componentSize = 0;
 		elektraFree (graph->edges);
 		elektraFree (graph);
 		return NULL;
 	}
-	size_t * removeSequenceNextEdge = elektraMalloc ((n + n * opmphm->rUniPar) * sizeof (size_t));
-	if (!removeSequenceNextEdge)
+	/* one malloc for:
+	 * - graph->removeSequence	n
+	 * - graph->edges[i].vertices	n * opmphm->rUniPar
+	 * - graph->edges[i].nextEdge	n * opmphm->rUniPar
+	 */
+	uint32_t * removeSequenceVerticesNextEdge = elektraMalloc ((n + 2 * n * opmphm->rUniPar) * sizeof (uint32_t));
```

Comment:
Is it possible to reuse previously allocated memory? What is the overhead of these allocations?

Reply:
Sure, it would be possible, but would cost much memory `O(n (r2 + c + 1))` for a `n > 240` r and c are `r=3` and `1.35 <= c <= 2.25`, to carry around for each KeySet.